### ISSUE TITLE
[UI Components] Replace `??` with the `||` operator in CloseButton

### ISSUE THREAD

**Issue by huyenltnguyen** [freeCodeCamp/freeCodeCamp#53987]
## Description

I got a mini heart attack as I randomly tested the Storybook page of the `CloseButton` component and found that the component failed a11y checks.

<img width="1680" alt="Screenshot 2024-03-06 at 00 49 32" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/25715018/782565a4-a41f-47a1-bb24-25ae7dd309e4">

This is because:
- We pass `label` as an empty string to the component in the Storybook demo:
   https://github.com/freeCodeCamp/freeCodeCamp/blob/19be14b72f33b914a8304683b6ae9e1f82599b05/tools/ui-components/src/close-button/close-button.stories.tsx#L14
- We use the `??` operator in the computation of the button's label, which would only fall back to `'Close'` if `label` is either `null` or `undefined`:
    https://github.com/freeCodeCamp/freeCodeCamp/blob/19be14b72f33b914a8304683b6ae9e1f82599b05/tools/ui-components/src/close-button/close-button.tsx#L51

Note that this issue is only in Storybook and not in /learn as we do pass a `label` to `CloseButton` there:

https://github.com/freeCodeCamp/freeCodeCamp/blob/19be14b72f33b914a8304683b6ae9e1f82599b05/client/src/components/Flash/index.tsx#L38-L42

## Changes needed

- In `close-button.tsx`, replace the `??` operator with the `||` operator
- In `close-button.stories.tsx`, make `args` an empty object instead of containing empty `className` and `label`

**Comment by ShaunSHamilton** [comment]
I do not care much about the implementation, BUT, surely if a label of `''` is never valid, then it is best to make the `label` optional, and not pass one to the component?

That is, I can imagine passing `label: ''` to a component because I do not want one, but was forced to pass one because it is not optional.
